### PR TITLE
Fix server chat messages being send twice when watching a replay

### PR DESCRIPTION
**Pull request by NoCatt**
Currently when watching a kill replay, all chat messages that are send from the server to the killer will be shown in the replay.

After this change all messages send will check if the game is currently replaying, and the message will not be shown in chat again.
This will avoid issues especially with messages only meant for the killer.

### CODE REVIEW

**Review by emma-miler — CHANGES_REQUESTED** [review]
This isn't actually stopping anything from being sent, only stops the client from showing the message received. Still useful, but not as advertised :P

**Review by uniboi — APPROVED** [review]
better than nothing

previous behaviour:
https://user-images.githubusercontent.com/64006268/209240022-90addfe0-fb56-4a20-8164-eb923ad5948c.mp4

new behaviour:
https://user-images.githubusercontent.com/64006268/209240031-29b31b5c-2caa-44f9-88c3-717d8a2f3f9f.mp4

**Comment by NoCatt** [comment]
> This isn't actually stopping anything from being sent, only stops the client from showing the message received. Still useful, but not as advertised :P

fixed :)
(I meant send to chat not send at all but its more clear now^^)

**Review by emma-miler — APPROVED** [review]

**Comment by cpdt** [comment]
Won’t this also prevent normal chat messages sent during replay from being received?

**Review by ASpoonPlaysGames — CHANGES_REQUESTED** [review]
> Won’t this also prevent normal chat messages sent during replay from being received?

Yes, which would result in the player missing messages entirely. Imo this is worse than receiving duplicate messages, so I have to disapprove of this PR for now

**Comment by NoCatt** [comment]
> > Won’t this also prevent normal chat messages sent during replay from being received?
> 
> Yes, which would result in the player missing messages entirely. Imo this is worse than receiving duplicate messages, so I have to disapprove of this PR for now

That is already the case though

We specifically checked that

**Comment by ASpoonPlaysGames** [comment]
> > > Won’t this also prevent normal chat messages sent during replay from being received?
> > 
> > 
> > Yes, which would result in the player missing messages entirely. Imo this is worse than receiving duplicate messages, so I have to disapprove of this PR for now
> 
> That is already the case though
> 
> We specifically checked that
> 
> So effectively nothing changes, but the double messages

Alright then, nevermind. Still I think that we should figure out how vanilla solves this problem, but this will do for now then

**Review by cpdt — APPROVED** [review]
LGTM then, thanks for clarifying.

**Review by ASpoonPlaysGames — APPROVED** [review]
This is a good enough temporary fix, eventually we really should get a better system for this though

File: Northstar.Client/mod/scripts/vscripts/_custom_codecallbacks_client.gnut [hunk 1]
```diff
@@ -22,6 +22,10 @@ struct {
 } NsCustomCallbacksClient
 
 void function OnReceivedMessage(ClClient_MessageStruct localMessage) {
+
+    if( IsWatchingReplay() )
```
Comment (Erlite) [inline Northstar.Client/mod/scripts/vscripts/_custom_codecallbacks_client.gnut:26]: Missing a space after `if`

**Review by Erlite — CHANGES_REQUESTED** [review]

File: Northstar.Client/mod/scripts/vscripts/_custom_codecallbacks_client.gnut (hunk 1 above)
Comment (NoCatt) [reply Northstar.Client/mod/scripts/vscripts/_custom_codecallbacks_client.gnut]: done

**Review by Erlite — APPROVED** [review]
Looks good to me, merge when ready. Bandaid that can be improved later in native.

**Review by BobTheBob9 — APPROVED** [review]
echoing same statements as previous reviewers, should work for now but needs changing later

**Pull request merged**